Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -0,0 +1,422 @@
+                    {
+                        ++size;
+                    }
+                    ++size;
+                }
+            }
+        }
+        return size;
+    }
+
+    void Path::FormatString(char* stringBuffer, size_t bufferSize) const
+    {
+        size_t bufferIndex = 0;
+
+        auto putChar = [&](char c)
+        {
+            if (bufferIndex == bufferSize)
+            {
+                return;
+            }
+            stringBuffer[bufferIndex++] = c;
+        };
+
+        auto writeToBuffer = [&](const char* key)
+        {
+            for (size_t keyIndex = 0; key[keyIndex]; ++keyIndex)
+            {
+                const char c = key[keyIndex];
+                if (c == EscapeCharacter)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(TildeSequence);
+                }
+                else if (c == PathSeparator)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(ForwardSlashSequence);
+                }
+                else
+                {
+                    putChar(c);
+                }
+            }
+        };
+
+        for (const PathEntry& entry : m_entries)
+        {
+            putChar(PathSeparator);
+            if (entry.IsEndOfArray())
+            {
+                putChar(EndOfArrayCharacter);
+            }
+            else if (entry.IsIndex())
+            {
+                bufferIndex += sprintf_s(&stringBuffer[bufferIndex], bufferSize - bufferIndex, "%zu", entry.GetIndex());
+            }
+            else
+            {
+                writeToBuffer(entry.GetKey().GetCStr());
+            }
+        }
+
+        putChar('\0');
+    }
+
+    AZStd::string Path::ToString() const
+    {
+        AZStd::string formattedString;
+        const size_t size = GetStringLength();
+        formattedString.resize(size);
+        FormatString(formattedString.data(), size + 1);
+        return formattedString;
+    }
+
+    void Path::FromString(AZStd::string_view pathString)
+    {
+        m_entries.clear();
+        size_t pathIndex = 0;
+        bool isNumber = true;
+        for (size_t i = 0; i <= pathString.size(); ++i)
+        {
+            if (i == pathString.size() || pathString[i] == PathSeparator)
+            {
+                if (i > 0)
+                {
+                    AZStd::string_view section(&pathString.data()[pathIndex], i - pathIndex);
+                    if (section.size() == 1 && section[0] == EndOfArrayCharacter)
+                    {
+                        PathEntry entry;
+                        entry.SetEndOfArray();
+                        m_entries.push_back(AZStd::move(entry));
+                    }
+                    else if (isNumber && section.size() > 0)
+                    {
+                        size_t index = 0;
+                        sscanf_s(section.data(), "%zu", &index);
```

Comment:
`scanf` is sign of code smell.

We should use strtoull for this

Reply:
I wasn't big on `scanf` either tbh - is it safe to use `strtoull` for `size_t`? I suppose it is for all of our current platforms, which is probably good enough.

Reply:
We only support 64-bit platforms, so `size_t` and `unsigned long long` have the same width(Not necessarily type though!)

Reply:
👍 , just always makes me a little nervous, like using `%llu` for it in an f-string.

Reply:
Since we lack any support for stringstreams which would normally make any such string to type conversions safe and easy, we've kind of built out a weak substitute in our AZ::Console system that you can take advantage of.
```
        template <typename _TYPE>
        inline bool StringToValue(_TYPE& outValue, AZStd::string_view string)
```

Reply:
bottom of this file:
https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Console/ConsoleTypeHelpers.inl

Reply:
Oh nice, I'll happily use that - we might want to look at moving that into our string API somewhere long-term if we're going to suggest other systems like this use it though.